### PR TITLE
Sidebar: Foundation

### PR DESCRIPTION
**Pull request by kean**
This is the initial PR adding the minimum required infrastructure for supporting the sidebar on iPad.

- These changes are intended to **not** affect the iPhone version in any way
- The changes are under the new `.sidebar` feature flag

None of the UI/UX introduced in this PR is final. I'll continue working on it on the screen-by-screen basis in the future PRs with the two main goals:

- (User-facing) Replace a tab bar on iPad with a sidebar as these tab bars are no longer supported on iPadOS 18
- (Technical) Remove `WPSplitViewController` and all the associated complexity

I will avoid making any other changes for now.

## To test:

Since none of the change are final, the main priority for testing is to ensure the iPhone is un-affected by this change, and the iPad version continues to work as is when the FF is off.

https://github.com/user-attachments/assets/552eaeac-0cd6-46e7-bcd7-8dfb78ee31e7

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/Classes/System/RootViewPresenter+MeNavigation.swift [hunk 1]
```diff
@@ -2,11 +2,6 @@ import UIKit
 
 /// Methods to access the Me Scene and sub levels
 extension RootViewPresenter {
-    /// removes all but the primary viewControllers from the stack
-    func popMeTabToRoot() {
```
Comment (kean) [inline WordPress/Classes/System/RootViewPresenter+MeNavigation.swift:6]: Unused code.

File: WordPress/Classes/ViewRelated/Blog/Blog Details/Detail Header/BlogDetailHeaderView.swift [hunk 2]
```diff
@@ -54,8 +54,7 @@ class BlogDetailHeaderView: UIView {
     @objc func refreshIconImage() {
         guard let blog else { return }
 
-        var viewModel = SiteIconViewModel(blog: blog)
-        viewModel.background = Color(.systemBackground)
+        let viewModel = SiteIconViewModel(blog: blog)
```
Comment (kean) [inline WordPress/Classes/ViewRelated/Blog/Blog Details/Detail Header/BlogDetailHeaderView.swift:57]: It's now passed using `@Environment`, which seems to be a better option.

**Comment by crazytonyli** [comment]
25.3 is scheduled to enter code freeze this week. Do you think we should avoid merging this PR until that's done?

**Comment by kean** [comment]
> 25.3 is scheduled to enter code freeze this week. Do you think we should avoid merging this PR until that's done?

I was careful to not intorduce any changes in the existing code, but I agree it would be safer to wait until the release branch is cut.

File: WordPress/Classes/System/SplitViewRootPresenter.swift [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+    }
+
+    // MARK: – RootViewPresenter
+
+    var rootViewController: UIViewController { splitVC }
+
+    var currentViewController: UIViewController?
+
+    func showBlogDetails(for blog: Blog) {
+        fatalError()
```
Comment (crazytonyli) [inline WordPress/Classes/System/SplitViewRootPresenter.swift:108]: Is calling the `showDetails(for site: Blog)` sufficient here?

**Review by crazytonyli — APPROVED** [review]

File: WordPress/Classes/System/SplitViewRootPresenter.swift (hunk 3 above)
Comment (kean) [reply WordPress/Classes/System/SplitViewRootPresenter.swift:108]: I will be implementing this protocol separately. If it is, I'll combine these methods into one.

**Pull request merged**